libs/langchain/langchain/vectorstores/tiledb.py

```diff
@@ -359,16 +414,77 @@ def max_marginal_relevance_search(
         )
         return docs
 
+    @classmethod
+    def create(
+        cls,
+        index_uri: str,
+        index_type: str,
+        dimensions: int,
+        vector_type: np.dtype,
+        metadatas: bool = True,
+        config: Optional[Mapping[str, Any]] = None,
+    ) -> None:
+        tiledb_vs, tiledb = dependable_tiledb_import()
+        with tiledb.scope_ctx(ctx_or_config=config):
+            try:
+                tiledb.group_create(index_uri)
+            except tiledb.TileDBError as err:
+                raise err
+            group = tiledb.Group(index_uri, "w")
+            vector_index_uri = f"{group.uri}/{VECTOR_INDEX_NAME}"
+            docs_uri = f"{group.uri}/{DOCUMENTS_ARRAY_NAME}"
+            if index_type == "FLAT":
+                tiledb_vs.flat_index.create(
+                    uri=vector_index_uri,
+                    dimensions=dimensions,
+                    vector_type=vector_type,
+                    config=config,
+                )
+            elif index_type == "IVF_FLAT":
+                tiledb_vs.ivf_flat_index.create(
+                    uri=vector_index_uri,
+                    dimensions=dimensions,
+                    vector_type=vector_type,
+                    config=config,
+                )
+            group.add(vector_index_uri, name=VECTOR_INDEX_NAME)
+
+            dim = tiledb.Dim(
```

Comment:
Let's make a note to add this document store as a feature in TileDB-Vector-Search ASAP so that we don't have to hard-code creation here.

Reply:
Done